sync/class.jetpack-sync-client.php

```diff
@@ -604,6 +631,52 @@ function filter_post_content_and_add_links( $post ) {
 		return $post;
 	}
 
+
+	function expand_wp_comment_status_change( $args ) {
+		return array( $args[0], $this->filter_comment_and_add_hc_meta( $args[1] ) );
+	}
+
+	function expand_wp_insert_comment( $args ) {
+		return array( $args[0], $this->filter_comment_and_add_hc_meta( $args[1] ) );
+	}
+
+	function filter_comment_and_add_hc_meta( $comment ) {
+		/**
+		 * Filters whether to prevent sending comment data to .com
+		 *
+		 * Passing true to the filter will prevent the comment data from being sent
+		 * to the WordPress.com.
+		 * Instead we pass data that will still enable us to do a checksum against the
+		 * Jetpacks data but will prevent us from displaying the data on in the API as well as
+		 * other services.
+		 * @since 4.2.0
+		 *
+		 * @param boolean false prevent post data from bing sycned to WordPress.com
+		 * @param mixed $comment WP_COMMENT object
+		 */
+		if ( apply_filters( 'jetpack_sync_prevent_sending_comment_data', false, $comment ) ) {
+			$blocked_comment = new stdClass();
+			$blocked_comment->comment_ID = $comment->comment_ID;
+			$blocked_comment->comment_date = $comment->comment_date;
+			$blocked_comment->comment_date_gmt = $comment->comment_date_gmt;
+			$blocked_comment->comment_approved = 'jetpack_sync_blocked';
+			return $blocked_comment;
+		}
+		
+		// add meta-property with Highlander Comment meta, which we 
+		// we need to process synchronously on .com
+		$hc_post_as = get_comment_meta( $comment->comment_ID, 'hc_post_as', true );
```

Comment:
This code should be removed. We no longer get highlander meta this way on the server - we use the synced meta instead.